lib/public/views/Runs/format/formatRunDetectors.js

```diff
@@ -29,23 +29,51 @@ export const formatRunDetectors = (run, detailsModel) => {
         return '-';
     }
 
+    const _detectorTypes = detailsModel.detectorTypes.match({
+        NotAsked: () => [],
+        Loading: () => [],
+        Success: (data) => data,
+        Failure: (error) => error,
+    });
+
+    const _allAndSelectedDetectorTypes = _detectorTypes.map(({ id, name }) => {
+        const isPresent = detectorsQualities.map((obj) => obj.id).includes(id);
+        const quality = isPresent ? detectorsQualities.filter((obj) => obj.id == id)[0].quality : null;
+        return { id, name, quality, isPresent };
+    });
```

Comment:
Be careful here: `_detectortypes` can be a list of errors.
What you should do instead is to wrap the whole component in the match `Success` case, because it does not makes sense in any other case.